deep_agent/aegra/middleware.py

```
@@ -37,8 +37,10 @@ def __init__(self, message: str, status_code: int = 401):
 def validate_api_key(provided_key: str) -> bool:
     """Constant-time comparison of API keys to prevent timing attacks."""
     if not API_KEY:
-        logger.warning("LANGGRAPH_API_KEY not set — all keys accepted")
-        return True
+        raise AuthError(
+            "LANGGRAPH_API_KEY is not configured; API key authentication is unavailable",
+            status_code=500,
+        )
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)